src/internal/connector/graph/uploadsession.go

```diff
@@ -48,17 +48,25 @@ func (iw *writer) Write(p []byte) (int, error) {
 
 	// PUT the request - set headers `Content-Range`to describe total size and `Content-Length` to describe size of
 	// data in the current request
-	_, err := iw.client.R().
-		SetHeaders(map[string]string{
-			contentRangeHeaderKey: fmt.Sprintf(
-				contentRangeHeaderValueFmt,
-				iw.lastWrittenOffset,
-				endOffset-1,
-				iw.contentLength),
-			contentLengthHeaderKey: fmt.Sprintf("%d", rangeLength),
-		}).
-		SetBody(bytes.NewReader(p)).
-		Put(iw.url)
+	req, err := http.NewRequest("PUT", iw.url, bytes.NewReader(p))
```

Comment:
It worries me that this is based off of the HTTPWrapper client, but then avoids calling its `Request()` func, which is the primary point of centralizing that struct.  Seems like the headers are the only standout conflict between the two uses?  What if you updated Request() to swap the `map[string]string` for a `http.Headers` value?  That should help you to work within the architecture.

Reply:
Make sense.